TicketBooking.Service/Services/BookingService/BookingService.cs

```diff
@@ -0,0 +1,215 @@
+                            FlightId = model.FlightId,
+                            FlightPrice = 0
+                        };
+                        var bookingList = mapper.Map<BookingList>(bookingListVM);
+                        await bookingListRepo.Add(bookingList);
+                    }
+                    else if (model.RoundFlightId != null)
+                    {
+                        var goFlight = new BookingListViewModel
+                        {
+                            NumberSeat = booking.NumberPeople,
+                            BookingId = booking.Id,
+                            FlightId = model.FlightId,
+                            FlightPrice = 0
+                        };
+                        var roundFlight = new BookingListViewModel
+                        {
+                            NumberSeat = booking.NumberPeople,
+                            BookingId = booking.Id,
+                            FlightId = model.RoundFlightId,
+                            FlightPrice = 0
+                        };
+                        var bookingGoFlight = mapper.Map<BookingList>(goFlight);
+                        var bookingRoundFlight = mapper.Map<BookingList>(roundFlight);
+                        booking.BookingLists?.Add(bookingGoFlight);
+                        booking.BookingLists?.Add(bookingRoundFlight);
+                        await bookingListRepo.Add(bookingGoFlight);
+                        await bookingListRepo.Add(bookingRoundFlight);
+                    }
+                    else
+                    {
+                        return new Response
+                        {
+                            Status = true,
+                            Message = "Booking failed, Invalid round trip flight",
+                            Data = booking
+                        };
+                    }
+                    await bookingRepo.AddBooking(booking);
+                    await unitOfWork.CompletedAsync();
+                    return new Response
+                    {
+                        Status = true,
+                        Message = "Booking success",
+                        Data = booking
+                    };
+                }
+                return new Response
+                {
+                    Status = true,
+                    Message = "Invalid flight ",
+                    Data = null
+                };
+            }
+            catch (InvalidOperationException)
+            {
+                throw new InvalidOperationException("Some propaties is valid !");
+            }
+            catch (Exception e)
+            {
+                throw new Exception(e.Message);
+            }
+
+        }
+
+        public async Task<Response> CancelBooking(Guid bookingId)
+        {
+            var booking = await bookingRepo.GetById(bookingId);
+            if (booking != null)
+            {
+                booking.Status = "Cancel";
```

Comment:
booking status should have enum or constant

Reply:
fixed